adaptors/dummy/test/e2e/usability/usability_test.go

```diff
@@ -0,0 +1,552 @@
+	v1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	apiv1alpha1 "github.com/rancher/octopus/adaptors/dummy/api/v1alpha1"
+	edgev1alpha1 "github.com/rancher/octopus/api/v1alpha1"
+	"github.com/rancher/octopus/pkg/util/object"
+	"github.com/rancher/octopus/test/util/content"
+	"github.com/rancher/octopus/test/util/exec"
+	"github.com/rancher/octopus/test/util/node"
+)
+
+var (
+	deviceLink edgev1alpha1.DeviceLink
+)
+
+var _ = Describe("verify usability", func() {
+
+	BeforeEach(func() {
+		// create device link
+		deployDummyDeviceLink()
+	})
+
+	AfterEach(func() {
+		// delete device link, ignore error
+		_ = k8sCli.DeleteAllOf(testCtx, &edgev1alpha1.DeviceLink{}, client.InNamespace(deviceLink.Namespace))
+	})
+
+	Context("edit dummy device link spec", func() {
+		Specify("if invalid node spec", func() {
+
+			By("given the device link is deployed", isDummyDeviceLinkDeployed)
+
+			By("when invalid node spec", invalidNodeSpec)
+
+			By("then the device link status should be NodeExisted false", isNodeExistedFalse)
+
+			By("when correct node spec", correctNodeSpec)
+
+			By("then the device link status should be NodeExisted true", isNodeExistedTrue)
+
+		})
+
+		Specify("if invalid model spec", func() {
+
+			By("given the device link is deployed", isDummyDeviceLinkDeployed)
+
+			By("when invalid model spec", invalidModelSpec)
+
+			By("then the device link status should be ModelExisted false", isModelExistedFalse)
+
+			By("when correct model spec", correctModelSpec)
+
+			By("then the device link status should be ModelExisted true", isModelExistedTrue)
+
+		})
+
+		Specify("if invalid adaptor spec", func() {
+
+			By("given the device link is deployed", isDummyDeviceLinkDeployed)
+
+			By("when invalid adaptor spec", invalidAdaptorSpec)
+
+			By("then the device link status should be AdaptorExisted false", isAdaptorExistedFalse)
+
+			By("when correct adaptor spec", correctAdaptorSpec)
+
+			By("then the device link status should be AdaptorExisted true", isAdaptorExistedTrue)
+
+		})
+
+		Specify("if invalid device spec", func() {
+
+			By("given the device link is deployed", isDummyDeviceLinkDeployed)
+
+			By("when invalid device spec", invalidDeviceSpec)
+
+			By("then the device link status should be DeviceConnected false", isDeviceConnectedFalse)
+
+			By("when correct device spec", correctDeviceSpec)
+
+			By("then the device link status should be DeviceConnected true", isDeviceConnectedTrue)
+
+		})
+
+		Specify("if invalid device spec to another valid spec", func() {
+
+			By("given the device link is deployed", isDummyDeviceLinkDeployed)
+
+			By("when edit spec.template.spec.gear to fast", editGearToFast)
+
+			By("then the device link work in config(fast)", isDeviceConfigFast)
+
+		})
+	})
+
+	Context("restart adaptors pods", func() {
```

Comment:
Why this context described `restart adaptor pods`?

Reply:
updated